### PR TITLE
IUSE tuning

### PR DESCRIPTION
**Pull request by a17r**
*Done:*

1) [kde-plasma/plasma-meta: Add IUSE unsupported, blocking bad packages](https://github.com/gentoo/kde/commit/fd2f3dc0286cde5f78ff93b78fec54748cbfb01d) 
For starters, blocking:
- gui-apps/qt6ct
- sys-apps/xdg-desktop-portal-gnome

2) [kde-plasma/plasma-meta: Add IUSE oxygen-theme for legacy style/icons/sounds](https://github.com/gentoo/kde/commit/d780b495efcb21fdea2b85277e9efadb28ae80c9) 
See also: https://community.kde.org/Distributions/Packaging_Recommendations#Plasma_packages

3) [kde-plasma/plasma-meta: Add IUSE +xwayland for xwaylandvideobridge](https://github.com/gentoo/kde/pull/981/commits/6f6ce64a63780f169b3451053525826ebbb550f6)

4) [kde-frameworks/kio: RDEPEND on dev-qt/qtbase:6[libproxy]](https://github.com/gentoo/kde/pull/981/commits/665a9dc3384bd16a5b2281a9aba007e73f4a932d) 
See also: https://community.kde.org/Plasma/Plasma_6.0_Release_notes#Everything_else
Upstream commit [22edf0fce94f5c2ed605d3636c1938e1b1c91987](https://invent.kde.org/frameworks/kio/-/commit/22edf0fce94f5c2ed605d3636c1938e1b1c91987)

5) [kde-plasma/plasma-meta: Add IUSE qt5 to trickle down integration USEdeps](https://github.com/gentoo/kde/pull/981/commits/f3557d0d0801834b92be7d0f2f19931066d0c75d)
USE=qt5 is currently globally enabled in desktop profiles.
\
Add a postinst warning coupled with the existence of dev-qt/qtgui.

**Abandoned:**

- [kde-plasma/plasma-meta: Add IUSE webengine for kde-apps/khelpcenter:[56]](https://github.com/gentoo/kde/pull/981/commits/98155ebfc6b33ef235ae262b72d7021013c220a7)
\
kde-apps/khelpcenter:5 uses kde-frameworks/khtml:5
kde-apps/khelpcenter:6 uses dev-qt/qtwebengine:6, unfortunately.
\
Plasma profile has so far not depended on dev-qt/webengine by default,
with the unfortunate porting away from deprecated khtml now it would.
\
This solution can obviously only be a temporary workaround and may
potentially cause some bewilderment by pulling in a KF5-based dependency
with default global settings "handbook -webengine".
\
Alternatively, we could put kde-apps/khelpcenter:6 behind IUSE webengine
directly, dropping the "handbook" relation. Application packages will
then by means of global "handbook" flag still generate full documentation.
\
At runtime, due to lack of installed khelpcenter binary, they seem to fall
back to using the default browser for opening - unfortunately - KDE's
online handbook instead of the locally installed one.

**Deferred:**

- [kde-apps/akonadi: Switch default from QMYSQL to QSQLITE, drop README](https://github.com/gentoo/kde/commit/54940d1ade9f4627b847fe105be87b0ae7e1d865) 
See also: https://mail.kde.org/pipermail/kde-pim/2023-April/049426.html
TODO: Migration for users? https://discuss.kde.org/t/migrate-akonadi-to-sqlite-from-postgres/7094
\
Reason: MYSQL is still default in git master.
https://invent.kde.org/pim/akonadi/-/blob/master/INSTALL?ref_type=heads#L37

---
TODO:
* bump `kde-frameworks/kf-env` to 6 to (among other things?) depend on `>=x11-misc/xdg-utils-1.2.0`
* problems with NOFILE limit? https://forums.gentoo.org/viewtopic-p-8812040.html#8812040
* problems with kscreenlocker pam config https://forums.gentoo.org/viewtopic-p-8815538.html#8815538?

### CODE REVIEW

**Review by ArsenArsen — APPROVED** [review]
all lgtm - quite happy to see sqlite gets love for desktop use over big rdbms'

File: kde-plasma/plasma-meta/plasma-meta-6.0.49.9999.ebuild [hunk 1]
```diff
@@ -80,12 +78,20 @@ RDEPEND="
 		>=kde-plasma/kde-gtk-config-${PV}:${SLOT}
 		x11-misc/appmenu-gtk-module
 	)
-	handbook? ( kde-apps/khelpcenter:* )
+	handbook? (
+		!webengine? ( kde-apps/khelpcenter:5 )
+		webengine? ( kde-apps/khelpcenter:6 )
+	)
```
Comment (ArsenArsen) [inline kde-plasma/plasma-meta/plasma-meta-6.0.49.9999.ebuild:84]: ah, I missed this on first pass.  perhaps we can get upstream behavior to be locally discovered manuals also

**Review by thesamesam — APPROVED** [review]
OK apart from sqlite where I don't know what we're going to do yet (I like the change, not the migration needed...)

**Comment by a17r** [comment]
Added `dev-qt/qtmultimedia:6[gstreamer]` cleanup re https://github.com/gentoo/gentoo/pull/35612

Although it means we lose `media-plugins/gst-plugins-soup`  dependency in case the user indeed selects gstreamer for qtmultimedia:6, a dependency that was added in order to fix https://bugs.gentoo.org/920833 in kde-plasma/kwin:5.

**Comment by ionenwks** [comment]
> Added `dev-qt/qtmultimedia:6[gstreamer]` cleanup re [gentoo/gentoo#35612](https://github.com/gentoo/gentoo/pull/35612)
> 
> Although it means we lose `media-plugins/gst-plugins-soup` dependency in case the user indeed selects gstreamer for qtmultimedia:6, a dependency that was added in order to fix https://bugs.gentoo.org/920833 in kde-plasma/kwin:5.

Kind of meh, but in that case could do:
```
|| (
    multimedia[ffmpeg]
    (
        multimedia[gstreamer]
        soup
    )
)
```

**Comment by ionenwks** [comment]
> > Added `dev-qt/qtmultimedia:6[gstreamer]` cleanup re [gentoo/gentoo#35612](https://github.com/gentoo/gentoo/pull/35612)
> > Although it means we lose `media-plugins/gst-plugins-soup` dependency in case the user indeed selects gstreamer for qtmultimedia:6, a dependency that was added in order to fix https://bugs.gentoo.org/920833 in kde-plasma/kwin:5.
> 
> Kind of meh, but in that case could do:
> 
> ```
> || (
>     multimedia[ffmpeg]
>     (
>         multimedia[gstreamer]
>         soup
>     )
> )
> ```

... not that it guarantee what is going to be used at runtime, I think it'll use ffmpeg if both are enabled unless go out of your way.

**Comment by thesamesam** [comment]
Yeah, let's try https://github.com/gentoo/kde/pull/981#issuecomment-1984824012.

**Pull request closed without merging**